backend/lucene/src/main/java/org/hibernate/search/backend/lucene/types/predicate/impl/LuceneExistsPredicate.java

```diff
@@ -58,7 +57,7 @@ private NormsBasedBuilder(LuceneSearchIndexScope<?> scope, LuceneSearchIndexValu
 
 		@Override
 		protected Query buildQuery() {
-			return new NormsFieldExistsQuery( absoluteFieldPath );
+			return new FieldExistsQuery( absoluteFieldPath );
```

Comment:
If this `FieldExistsQuery` always works regardless of the field type, maybe you can merge all the different builders/factories in this class?

Reply:
Yup, this makes sense - Lucene now have a huge if-else-switch covering these scenarios 🙈 will merge